app/tests.py

```diff
@@ -0,0 +1,50 @@
+from django.test import TestCase
+from app.forms import RegistrationForm, ProfilePasswordChangeForm
+
+
+class RegisterTestCase(TestCase):
+
+    def test_register_username_correct(self):
+        form_data = {
+            'username': 'Jeyabal@-1',
+            'password': '1234',
+            'date_of_birth': '1989-03-10',
+        }
+        form = RegistrationForm(data=form_data)
+        self.assertEqual(form.is_valid(), True)
+
+    def test_register_username_incorrect(self):
+        form_data = {
+            'username': 'Jeyabal#',
+            'password': '1234',
+            'date_of_birth': '1989-03-10',
+        }
+        form = RegistrationForm(data=form_data)
+        self.assertEqual(form.is_valid(), False)
+
+    def test_register_password_incorrect(self):
+        form_data = {
+            'username': 'Jeyabal#',
+            'password': '12345',
+            'date_of_birth': '1989-03-10',
+        }
+        form = RegistrationForm(data=form_data)
+        self.assertEqual(form.is_valid(), False)
+
+    def test_password_change_incorrect(self):
+        form_data = {
+            'old_password': '123',
+            'new_password': 'jey123',
+            'confirm_password': 'jey123',
+        }
+        form = ProfilePasswordChangeForm(data=form_data)
+        self.assertEqual(form.is_valid(), False)
+
+    def test_password_change_correct(self):
+        form_data = {
+            'old_password': '1234',
+            'new_password': 'jey3',
+            'confirm_password': 'jey3',
```

Comment:
These are invalid passwords according to the spec, I think they're only allowed to be 4 digits.